### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by annumberhocker**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,55 @@
-# IBM Cloud <module_name> - Terraform Module
+# IBM Cloud Pak for Integration - Terraform Module
```
Comment (umarali-nagoor) [inline README.md:1]: This comment is on Github actions

Looks like Github actions are failing ... please check

File: versions.tf [hunk 2]
```diff
@@ -3,28 +3,18 @@
...
 
-/***************************************************
-NOTE: To source a particular version of IBM terraform
-provider version, configure the parameter `version`.
-
 terraform {
-  required_version = ">=0.13"
+  required_version = ">= 0.13"
   required_providers {
     ibm = {
```
Comment (umarali-nagoor) [inline versions.tf:9]: Do we need ibm provider on module side ?

On example (template) side yes I agree we need ibm provider to read cluster_config _path but are we using ibm provider on module side as well ?

File: README.md [hunk 3]
```diff
@@ -50,49 +59,52 @@ module "logdna_instance" {
 ### Terraform plugins
 
 - [Terraform](https://www.terraform.io/downloads.html) 0.13 (or later)
```
Comment (umarali-nagoor) [inline README.md:61]: Add a sub-section "pre-requisites" to capture some of the pre-requisites like open-shift cluster etc

File: README.md [hunk 4]
```diff
@@ -1,46 +1,55 @@
...
+  config_dir        = local.kube_config_path
+}
+
+// Cloud Pak for Integration module
+module "cp4i" {
+  source = "../.."
+  enable = true
 
+  // ROKS cluster parameters:
+  cluster_config_path = data.ibm_container_cluster_config.cluster_config.config_file_path
```
Comment (umarali-nagoor) [inline README.md:45]: Add input & output variables table

File: README.md [hunk 5]
```diff
@@ -101,28 +113,32 @@ See upgrade guide at https://ibm.biz/detect-secrets-how-to-upgrade
 
 run below command
 
-```
+```bash
 pre-commit autoupdate
 ```
+
```
Comment (umarali-nagoor) [inline README.md:119]: Please capture the cp4i cleanup information

File: scripts/install_cp4i.sh [hunk 6]
```diff
@@ -0,0 +1,89 @@
+#!/bin/sh
+
+# Optional input parameters with default values:
```
Comment (umarali-nagoor) [inline scripts/install_cp4i.sh:3]: Scripts to uninstall cp4i ?

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,6 @@ All notable changes to this project will be documented in this file.
 Extending the adopted spec, each change should have a link to its
```
[the remark this answers is not in the document]
Comment (annumberhocker) [reply CHANGELOG.md:5]: I am not familiar with git workflows.   Can you go over what this all does so I know how to modify for CP 4 Int?

File: README.md (hunk 1 above)
Comment (annumberhocker) [reply README.md:1]: I do not understand this comment.  Where do I need to look to see what is failing?

File: CHANGELOG.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (annumberhocker) [reply CHANGELOG.md:5]: I added the identical lines to our test.yml file.

File: examples/cp4i/README.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
+# Example to provision CP4I Terraform Module
+
+## Run using IBM Cloud Schematics
+
+For instructions to run these examples using IBM Schematics go [here]([../Using_Schematics.md](https://cloud.ibm.com/docs/schematics?topic=schematics-get-started-terraform))
```
[the remark this answers is not in the document]
Comment (annumberhocker) [reply examples/cp4i/README.md:5]: Removed this section

File: examples/cp4i/main.tf [hunk 9]
```diff
@@ -0,0 +1,34 @@
+provider "ibm" {
```
[the remark this answers is not in the document]
Comment (annumberhocker) [reply examples/cp4i/main.tf:1]: removed this section

File: main.tf [hunk 10]
```diff
@@ -1,7 +1,58 @@
-/********************************************************************
+locals {
```
[the remark this answers is not in the document]
Comment (annumberhocker) [reply main.tf:1]: Yes, we determined it was type-2 during our discussions

File: versions.tf (hunk 2 above)
Comment (annumberhocker) [reply versions.tf:9]: We have found that sometimes there are problems if this isn't defined on the module side.

File: main.tf [hunk 11]
```diff
@@ -1,7 +1,58 @@
...
+  })
+  navigator_content = templatefile("${path.module}/templates/navigator.yaml.tmpl", {
+    storageclass = var.storageclass
+  })
+}
 
-This file is used to implement the ROOT module.
+# This section checks to see if the values have been updated through out the script running and is required for any dynamic value
+resource "null_resource" "install_cp4i" {
+  count = var.enable ? 1 : 0
```
[the remark this answers is not in the document]
Comment (annumberhocker) [reply main.tf]: I have removed this

File: examples/cp4i/README.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+For more information on IBM Schematics, refer [here](https://cloud.ibm.com/docs/schematics?topic=schematics-get-started-terraform).
+
+## Run using local Terraform Client
+
+For instructions to run using the local Terraform Client on your local machine go [here]([../Using_Terraform.md](https://ibm.github.io/cloud-enterprise-examples/iac/setup-environment/)).
```
[the remark this answers is not in the document]
Comment (annumberhocker) [reply examples/cp4i/README.md]: removed this section

File: README.md (hunk 3 above)
Comment (annumberhocker) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (annumberhocker) [reply README.md:56]: done

File: examples/cp4i/main.tf (hunk 9 above)
[the remark this answers is not in the document]
Comment (annumberhocker) [reply examples/cp4i/main.tf:1]: I have create a .go testcase

File: scripts/install_cp4i.sh (hunk 6 above)
Comment (annumberhocker) [reply scripts/install_cp4i.sh:3]: Added an uninstall_cp4i.sh

File: README.md (hunk 5 above)
Comment (annumberhocker) [reply README.md:140]: Created an uninstall_cp4i.sh file.  This will be run when `terraform destroy` is run.

**Comment by annumberhocker** [comment]
Need Umar to review changes

**Pull request merged**